internal/definitions/definition_handler_contracts.go

```diff
@@ -53,73 +53,79 @@ func (dh *definitionHandlers) persistFFI(ctx context.Context, ffi *fftypes.FFI)
 }
 
 func (dh *definitionHandlers) persistContractAPI(ctx context.Context, api *fftypes.ContractAPI) (valid bool, err error) {
-	err = dh.database.UpsertContractAPI(ctx, api, database.UpsertOptimizationSkip)
+	existing, err := dh.database.GetContractAPIByName(ctx, api.Namespace, api.Name)
 	if err != nil {
 		return false, err
 	}
-	return true, nil
+	if existing != nil && err == nil {
+		if !api.LocationAndLedgerEquals(existing) {
+			return false, nil
+		}
+	}
+	err = dh.database.UpsertContractAPI(ctx, api, database.UpsertOptimizationSkip)
```

Comment:
I think we could leverage `UpsertOptimizationInsert`, right? Generally this method will be the first time the API comes into existence?